src/react_native/cameraroll.cljs

```diff
@@ -0,0 +1,16 @@
+(ns react-native.cameraroll
+  (:require ["@react-native-community/cameraroll" :as CameraRoll]
+            [utils.transforms :as transforms]
+            [taoensso.timbre :as log]))
+
+(defn get-photos
+  [opts callback]
+  (-> (.getPhotos CameraRoll (clj->js opts))
+      (.then #(callback (transforms/js->clj %)))
```

Comment:
We should avoid using promises like this because we can pipeline over the values returned by promises. While doing so, it's much more common to name the argument `callback` as `promise`. Same feedback for the function `get-albums`.

```clojure
(defn get-photos
  [opts promise]
  (-> (.getPhotos CameraRoll (clj->js opts))
      (.then transforms/js-clj)
      (.then promise)
      (.catch #(log/warn "could not get camera roll photos" %))))
```

Reply:
are you sure, how does it work? `transforms/js-clj` won't return promise no ?

Reply:
Any function can be piped with promises, then don't need to be async, quite the contrary. That's why the abstraction is so flexible and powerful, because we can mix and match sync or async functions. As long as we return a value, it is passed along to the next function in the promise chain.

Reply:
idk, personally I feel like something is wrong here, mixing two worlds, i'm against promises , i would isolate them as much as possible, i would use `.then` only once which should handle clojurescript function, instead of mixing , so for me 
```clojure
(-> (promise-1)
      (.then clojure-world)
      (.catch)
```
looks more natural
IMHO

Reply:
I agree, my example is not the best one maybe, but this misconception in promise usage is the reason the events namespace is mixing nesting callbacks with promises. It could be much more clean, but only with code it would be clearer.